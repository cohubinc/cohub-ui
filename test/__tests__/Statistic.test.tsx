import React from "react";
import { mount } from "enzyme";

import { Statistic } from "dist";

describe("Statistic", () => {
  it("renders without crashing", async () => {
    const wrapper = mount(
      <Statistic label="test stat" value={100} format="number" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("render shopify formatted money correctly", () => {
    const wrapper = mount(
      <Statistic
        label="test stat"
        value={2344.5}
        format="money"
        shopCurrencyFormat="<h1 class=money>€ {{amount}} EUR</h1>"
      />
    );

    expect(wrapper.text()).toContain("€ 2,344.50 EUR");
    expect(wrapper.text()).not.toContain("<h1 class=money>€ 2,344.50</h1>");
  });
});
