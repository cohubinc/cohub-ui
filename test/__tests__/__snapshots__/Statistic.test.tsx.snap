// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Statistic
  format="number"
  label="test stat"
  value={100}
>
  <div
    className="flex flex-column justify-center items-center"
  >
    <Component
      color="hsl(0, 2%, 12%)"
      weight={600}
    >
      <span
        className=""
        data-qa="text"
        style={
          Object {
            "color": "hsl(0, 2%, 12%)",
            "fontFamily": "Inter",
            "fontSize": 16,
            "fontWeight": 600,
            "textAlign": "initial",
            "textTransform": "initial",
          }
        }
      >
        <FormatNumber
          thousandSeparator={true}
          value={100}
        >
          <NumberFormat
            allowEmptyFormatting={false}
            allowLeadingZeros={false}
            allowNegative={true}
            decimalSeparator="."
            displayType="text"
            fixedDecimalScale={false}
            getInputRef={[Function]}
            isAllowed={[Function]}
            isNumericString={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            onValueChange={[Function]}
            prefix=""
            suffix=""
            thousandSeparator={true}
            thousandsGroupStyle="thousand"
            type="text"
            value={100}
          >
            <span>
              100
            </span>
          </NumberFormat>
        </FormatNumber>
      </span>
    </Component>
    <Typography
      muted={true}
      uppercase={true}
    >
      <Component
        muted={true}
        uppercase={true}
      >
        <span
          className=""
          data-qa="text"
          style={
            Object {
              "color": "hsl(356, 3%, 59%)",
              "fontFamily": "Inter",
              "fontSize": "14px",
              "fontWeight": 400,
              "textAlign": "initial",
              "textTransform": "uppercase",
            }
          }
        >
          test stat
        </span>
      </Component>
    </Typography>
  </div>
</Statistic>
`;
